From: =?utf-8?q?Ond=C5=99ej_Sur=C3=BD?= <ondrej@sury.org>
Date: Fri, 13 May 2016 15:27:35 +0200
Subject: Require libical >= 2.0 and define the libical feature macros
 according to that

---
 configure.ac | 22 +++++++++-------------
 1 file changed, 9 insertions(+), 13 deletions(-)

diff --git a/configure.ac b/configure.ac
index 8e460d3..cfeb4de 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1209,7 +1209,7 @@ dnl
 	ENABLE_RSS=yes
 	ENABLE_DAV=yes
 	PKG_CHECK_MODULES([XML2], [libxml-2.0],,AC_MSG_ERROR([Need libxml-2.0 for http]))
-	PKG_CHECK_MODULES([ICAL], [libical],,AC_MSG_ERROR([Need libical for http]))
+	PKG_CHECK_MODULES([ICAL], [libical >= 2.0],,AC_MSG_ERROR([Need libical >= 2.0.0 for http]))
 	PKG_CHECK_MODULES([SQLITE3], [sqlite3],,AC_MSG_ERROR([Need sqlite3 for http]))
 	PKG_CHECK_MODULES([JSON], [jansson],,AC_MSG_ERROR([Need jansson for http]))
 
@@ -1227,9 +1227,8 @@ dnl
 		AC_DEFINE(HAVE_TZ_BY_REF,[],
 			[Build TZ by ref support into httpd?]))
 
-	AC_EGREP_HEADER(ICAL_TZUNTIL_PROPERTY, ical.h,
-		AC_DEFINE(HAVE_TZDIST_PROPS,[],
-			[Do we have built-in support for TZdist props?]))
+	AC_DEFINE(HAVE_TZDIST_PROPS,[],
+		[Do we have built-in support for TZdist props?])
 
 	CYRUS_OPENDKIM_CHK()
 	if test "$opendkimlib" = "yes"; then
@@ -1238,17 +1237,14 @@ dnl
 	    AC_DEFINE(WITH_DKIM,[],[Build DKIM support into iSchedule?]) 
 	fi
 
-	AC_EGREP_HEADER(ICAL_SCHEDULESTATUS_PARAMETER, ical.h,
-		AC_DEFINE(HAVE_SCHEDULING_PARAMS,[],
-			[Do we have built-in support for scheduling params?]))
+	AC_DEFINE(HAVE_SCHEDULING_PARAMS,[],
+		[Do we have built-in support for scheduling params?])
 
-	AC_EGREP_HEADER(ICAL_VAVAILABILITY_COMPONENT, ical.h,
-		AC_DEFINE(HAVE_VAVAILABILITY,[],
-			[Build VAVAILABILITY support into httpd?]))
+	AC_DEFINE(HAVE_VAVAILABILITY,[],
+		[Build VAVAILABILITY support into httpd?])
 
-	AC_EGREP_HEADER(ICAL_VPOLL_COMPONENT, ical.h,
-		AC_DEFINE(HAVE_VPOLL,[],
-			[Build VPOLL support into httpd?]))
+	AC_DEFINE(HAVE_VPOLL,[],
+		[Build VPOLL support into httpd?])
 
 	ENABLE_RSCALE=yes
 	PKG_CHECK_MODULES([ICU], [icu-i18n],, ENABLE_RSCALE=no)
