From: Sven Mueller <sven@debian.org>
Date: Tue, 4 Aug 2015 10:38:06 +0200
Subject: Change the wording of sieved's notice when a user's default sieve script is missing

---
 sieve/script.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/sieve/script.c b/sieve/script.c
index 7babef6..6ec838a 100644
--- a/sieve/script.c
+++ b/sieve/script.c
@@ -51,6 +51,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <errno.h>
 
 #include "assert.h"
 #include "charset.h"
@@ -431,7 +432,11 @@ EXPORTED int sieve_script_load(const char *fname, sieve_execute_t **ret)
     if (!fname || !ret) return SIEVE_FAIL;
 
     if (stat(fname, &sbuf) == -1) {
-        syslog(LOG_DEBUG, "IOERROR: fstating sieve script %s: %m", fname);
+        if (errno == ENOENT) {
+                syslog(LOG_DEBUG, "WARNING: sieve script %s doesn't exist: %m", fname);
+        } else {
+                syslog(LOG_DEBUG, "IOERROR: fstating sieve script %s: %m", fname);
+        }
         return SIEVE_FAIL;
     }
 
