From: =?utf-8?q?Ond=C5=99ej_Sur=C3=BD?= <ondrej@sury.org>
Date: Mon, 21 Sep 2015 16:44:19 +0200
Subject: missing-AC_LANG_SOURCE

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- cyrus-imapd.orig/configure.ac
+++ cyrus-imapd/configure.ac
@@ -530,11 +530,11 @@ if test "$spt_type" = ""; then
 fi
 if test "$spt_type" = ""; then
         AC_MSG_CHECKING(for PS_STRINGS)
-AC_CACHE_VAL(cyrus_cv_sys_psstrings, AC_COMPILE_IFELSE([
+AC_CACHE_VAL(cyrus_cv_sys_psstrings, AC_COMPILE_IFELSE([AC_LANG_SOURCE([
 #include <machine/vmparam.h>
 #include <sys/exec.h>
 void *p = PS_STRINGS;
-],cyrus_cv_sys_psstrings=yes,cyrus_cv_sys_psstrings=no))
+])],cyrus_cv_sys_psstrings=yes,cyrus_cv_sys_psstrings=no))
          if test $cyrus_cv_sys_psstrings = yes; then
                  spt_type=SPT_PSSTRINGS
          fi
