From: =?utf-8?q?Ond=C5=99ej_Sur=C3=BD?= <ondrej@sury.org>
Date: Tue, 22 Sep 2015 13:38:04 +0200
Subject: Fix typo in sphinx that disabled squat

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 4e143a2..5bd4877 100644
--- a/configure.ac
+++ b/configure.ac
@@ -370,7 +370,7 @@ dnl
 dnl Search engines - Sphinx
 dnl
 
-enable_squat=no
+enable_sphinx=no
 AC_ARG_ENABLE(sphinx,
         [  --enable-sphinx         disable Sphinx support],[
             enable_sphinx=$enableval
