--- cyrus-imapd.orig/lib/imapoptions
+++ cyrus-imapd/lib/imapoptions
@@ -1417,12 +1417,13 @@ If all partitions are over that limit, t
    in minutes.  The minimum value is 10, the default. */
 
 { "popuseacl", 0, SWITCH }
+
 /* Enforce IMAP ACLs in the pop server.  Due to the nature of the POP3
-   protocol, the only rights which are used by the pop server are 'r',
-   't', and 's' for the owner of the mailbox.  The 'r' right allows the
-   user to open the mailbox and list/retrieve messages.  The 't' right
-   allows the user to delete messages.  The 's' right allows messages
-   retrieved by the user to have the \\Seen flag set (only if
+   protocol, the only rights which are used by the pop server are 'r', 't',
+   and 's' for the owner of the mailbox.  The 'r' right allows
+   the user to open the mailbox and list/retrieve messages.  The 't'
+   right allows the user to delete messages.  The 's' right allows
+   messages retrieved by the user to have the \\Seen flag set (only if
    \fIpopuseimapflags\fR is also enabled). */
 
 { "popuseimapflags", 0, SWITCH }
--- cyrus-imapd.orig/man/cyr_dbtool.8
+++ cyrus-imapd/man/cyr_dbtool.8
@@ -85,8 +85,8 @@ terminated by tab or newline, values are
 Running without any options will list the available database backends and
 usable actions.
 .PP
-The 'consistency' command runs a consistency check on the DB by calling
-'myconsistent' on it.
+The 'consistency' command runs a consistency check on the DB by
+calling 'myconsistent' on it.
 .PP
 The 'repack' command will compress the database by removing stale data
 on backends which support it.  It's a NOOP otherwise.
