--- cyrus-imapd.orig/master/master.c
+++ cyrus-imapd/master/master.c
@@ -70,6 +70,10 @@
 #include <limits.h>
 #include <math.h>
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 #ifndef INADDR_NONE
 #define INADDR_NONE 0xffffffff
 #endif
--- cyrus-imapd.orig/master/service.c
+++ cyrus-imapd/master/service.c
@@ -72,6 +72,10 @@
 #include "strarray.h"
 #include "signals.h"
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
 extern int optind, opterr;
 extern char *optarg;
 
--- cyrus-imapd.orig/imap/pop3d.c
+++ cyrus-imapd/imap/pop3d.c
@@ -96,6 +96,10 @@
 
 #include "iostat.h"
 
+#ifndef MAXHOSTNAMELEN
+#define MAXHOSTNAMELEN
+#endif
+
 #ifdef HAVE_KRB
 /* kerberos des is purported to conflict with OpenSSL DES */
 #define DES_DEFS
