From 6be0fcb9f463091bec1abd86e3ba1cc9317ed028 Mon Sep 17 00:00:00 2001
From: Ken Murchison <murch@andrew.cmu.edu>
Date: Sun, 17 Apr 2011 18:56:45 +0000
Subject: Fixed bug #3444: properly reset the end of the string when searching for

supported SASL mechs in IMAP capability response
---
--- a/imtest/imtest.c
+++ b/imtest/imtest.c
@@ -1646,7 +1646,7 @@ static char *imap_parse_mechlist(const c
 	strlcat(ret, tmp, strlen(ret) + (end - tmp) + 1);
 	
 	/* reset the string */
-	str = end + 1;
+	str = end;
     }
     
     return ret;
